client/src/components/PacmanGame/PacmanGame.jsx

```diff
@@ -67,17 +73,62 @@ class PacmanGame extends Component {
     document.addEventListener('keydown', this.setDirection);
   };
 
-  moveGhosts = () => {
-    const { ghosts, gridState } = this.state;
-    const ghostUpdated = ghosts.map(({ x, y, direction }) => {
-      const newGhostLocation = getRandomAdjacentAvailableCell(gridState, { x, y, direction });
+  addPositionsToArray = (arr, index) => {
+    const scatterTime = 55;
+    if (arr.length < scatterTime) {
+      arr.push(boardCorners[index]);
+      return this.addPositionsToArray(arr, index);
+    }
+    return arr;
+  }
 
-      return newGhostLocation;
-    });
-    return {
-      gridState, ghosts: ghostUpdated,
+    moveGhosts = () => {
+      const { ghosts, gridState, scatterStart } = this.state;
+      let { moveGhostsCount, scatterGhostspath } = this.state;
+      moveGhostsCount += 1;
+      const scatterEnd = scatterStart + 55;
+      if (moveGhostsCount === scatterStart) {
+        const gridWithWeights = getGridwithWeights(gridState);
+        const ghostsPath = ghosts
+          .map((ghost, index) => chaseLocation(gridWithWeights, ghost, boardCorners[index]))
+          .map((postion) => {
+            const positionObject = postion.map(arr => ({ x: arr[0], y: arr[1] }));
+            return positionObject;
+          });
+        scatterGhostspath = ghostsPath.map((array, index) => {
+          const newArray = this.addPositionsToArray(array, index);
+          return newArray;
+        });
+        this.setState({
+          scatterGhostspath,
+        });
+      }
+
+      if (moveGhostsCount === scatterEnd) {
+        const ghostsUpdated = [{ x: 1, y: 1, direction: 'LEFT' },
+          { x: 23, y: 1, direction: 'LEFT' }, { x: 1, y: 23, direction: 'LEFT' },
+          { x: 23, y: 23, direction: 'LEFT' }];
+        return {
+          gridState, ghosts: ghostsUpdated, moveGhostsCount,
+        };
+      }
+
+      if (moveGhostsCount > (scatterStart + 1) && moveGhostsCount < scatterEnd) {
+        const ghostsUpdated = scatterGhostspath.map(path => path[moveGhostsCount - scatterStart]);
+        return {
+          gridState, ghosts: ghostsUpdated, moveGhostsCount,
+        };
+      }
+
+      const ghostUpdated = ghosts.map(({ x, y, direction }) => {
+        const newGhostLocation = getRandomAdjacentAvailableCell(gridState, { x, y, direction });
```

Comment:
Why a variable you can directly return getRandomAdjacentAvailableCell(gridState, { x, y, direction }).